src/api/procedures/leaveIdentity.ts

```diff
@@ -20,15 +20,20 @@ export async function prepareLeaveIdentity(
     context: {
       polymeshApi: { tx },
     },
-    context,
   } = this;
-  const {
-    account: { address },
-  } = args;
+  const { account } = args;
+
+  const currentIdentity = await account.getIdentity();
 
-  const currentIdentity = await context.getCurrentIdentity();
+  if (!currentIdentity) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'There is no Identity associated to this Account',
+    });
+  }
```

Comment:
This procedure should require that account is current Account too. Otherwise it could generate misleading behaviors